### PR TITLE
Add example-workflows

### PR DESCRIPTION
**Pull request by highway7**
Added an example workflow of Github Actions for deploying a python script to Google Cloud Functions.

### CODE REVIEW

File: example-workflows/.github/workflows/cloud-functions.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+name: Build and Deploy to Cloud Functions
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  # GCP_PROJECT is defined in service_account_key
+  PROJECT_ID: ${{ secrets.GCP_PROJECT }}
```
Comment (bharathkkb) [inline example-workflows/.github/workflows/cloud-functions.yml:24]: Can we remove `PROJECT_ID` as we dont use it in the workflow?

File: example-workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    represent best practices around the Principle of Least Privilege. To
+    properly restrict access, you should create a custom IAM role with the most
+    restrictive permissions.
+
+
+1.  [Create a JSON service account key][create-key] for the service account.
+
+1.  Add the following secrets to your repository's secrets:
+
+    - `RUN_SA_KEY`: the content of the service account JSON file
```
Comment (bharathkkb) [inline example-workflows/README.md:88]: Should this be `gcp_credentials` as per the workflow?

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @highway7 and apologies for the delay.
Some minor suggestions below

File: .gitignore [hunk 3]
```diff
@@ -1,45 +1,87 @@
-node_modules/
-runner/
+# Copyright 2019 Google LLC
```
Comment (bharathkkb) [inline .gitignore:1]: Looks like this was added in https://github.com/google-github-actions/deploy-cloud-functions/pull/20/commits/3d1f1ef3620a2221ecf9665645f578cb0918b4d2
We should keep the existing gitignore and only append if needed.

File: example-workflows/.github/workflows/cloud-functions.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+name: Build and Deploy to Cloud Functions
+
+on:
+  push:
+    branches:
+    - main
+
+env:
+  FUNCTION_REGION: northamerica-northeast1
```
Comment (bharathkkb) [inline example-workflows/.github/workflows/cloud-functions.yml:23]: We don't seem to be using this either? You will need to pass this in via `region` in the action or let it default to `us-central1`

**Review by bharathkkb — COMMENTED** [review]
Overall LGTM
/cc @averikitsch for any suggestions

File: example-workflows/.github/workflows/cloud-functions.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 Google, LLC.
```
Comment (averikitsch) [inline example-workflows/.github/workflows/cloud-functions.yml:1]: Update year

File: example-workflows/.github/workflows/cloud-functions.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - id: Deploy
+      uses: google-github-actions/deploy-cloud-functions@main
+      with:
+        name: helloworld-python
+        runtime: python37
+        entry_point: hello_world
+        credentials: ${{ secrets.gcp_credentials }}
```
Comment (averikitsch) [inline example-workflows/.github/workflows/cloud-functions.yml:40]: nit: for consistency use
```suggestion
        credentials: ${{ secrets.GCP_SA_KEY }}
```

File: example-workflows/.github/workflows/cloud-functions.yml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+env:
+  FUNCTION_REGION: northamerica-northeast1
+
+jobs:
+  setup-build-deploy:
+    name: Setup, Build, and Deploy
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (averikitsch) [inline example-workflows/.github/workflows/cloud-functions.yml:30]: It also may be nice to add a step to curl the output URL

File: example-workflows/README.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+4.  Run google-github-actions/deploy-cloud-functions
+    - Setting project Id from $GCLOUD_PROJECT
+    - Creating a function revision
+    - Creating or Updating function deployment
+
+
+## Setup
+
+1.  Create a new Google Cloud Project (or select an existing project) and
+    [enable the Cloud Functions and Cloud Build APIs](https://console.cloud.google.com/flows/enableapi?apiid=cloudbuild.googleapis.com,run.googleapis.com).
```
Comment (averikitsch) [inline example-workflows/README.md:45]: Is Cloud Build needed?

Comment (averikitsch) [inline example-workflows/README.md:45]:
```suggestion
    [enable the Cloud Functions and Cloud Build APIs](https://console.cloud.google.com/flows/enableapi?apiid=cloudbuild.googleapis.com,cloudfunctions.googleapis.com).
```

File: example-workflows/README.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+1.  Create or reuse a GitHub repository for the example workflow:
```
Comment (averikitsch) [inline example-workflows/README.md:47]: I'm not a fan of the create a repo and copy format. That doesn't make this a quick example. In the other repos the [example workflows](https://github.com/google-github-actions/deploy-cloudrun#example-workflows) still live in the `.github/workflows` directory with a specific branch push to trigger them.

File: example-workflows/README.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    - `Cloud Build Service Account` - allows for deploying cloud builds
+
+    - `Viewer` - allows for viewing the project
+
+    - `Service Account User` -  required to deploy services to Cloud Run
+
+    Note: These permissions are overly broad to favor a quick start. They do not
+    represent best practices around the Principle of Least Privilege. To
+    properly restrict access, you should create a custom IAM role with the most
+    restrictive permissions.
```
Comment (averikitsch) [inline example-workflows/README.md:81]: These don't seem necessary (https://github.com/google-github-actions/deploy-cloud-functions#authorization)

File: example-workflows/README.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
+# Cloud Functions - GitHub Actions
+
+An example [workflow](.github/workflows/cloud-functions.yml) that uses [GitHub Actions][actions] to deploy a
+[Hello World Python app](main.py) to [Cloud Funtions][cloud-functions].
+
+This code is intended to be an _example_. You will likely need to change or
+update values to match your setup.
+
+## Workflow description
```
Comment (averikitsch) [inline example-workflows/README.md:9]: Please review all steps, these don't match the workflow file.

**Review by averikitsch — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (highway7) [reply .gitignore:1]: Sorry it was my bad. Fixed by reverting it back.

File: example-workflows/.github/workflows/cloud-functions.yml (hunk 4 above)
Comment (highway7) [reply example-workflows/.github/workflows/cloud-functions.yml]: Corrected by removing it and using default.

File: example-workflows/.github/workflows/cloud-functions.yml (hunk 5 above)
Comment (highway7) [reply example-workflows/.github/workflows/cloud-functions.yml]: Updated with 2021.

File: example-workflows/.github/workflows/cloud-functions.yml (hunk 6 above)
Comment (highway7) [reply example-workflows/.github/workflows/cloud-functions.yml]: Changed to GCP_SA_KEY for consistency.

File: example-workflows/.github/workflows/cloud-functions.yml (hunk 7 above)
Comment (highway7) [reply example-workflows/.github/workflows/cloud-functions.yml:27]: Added a new step of testing with the output URL.

File: example-workflows/README.md (hunk 8 above)
Comment (highway7) [reply example-workflows/README.md]: Removed Cloud Build APIs since it is not needed.

Comment (highway7) [reply example-workflows/README.md]: Removed Cloud Run APIs since it is not needed.

File: example-workflows/README.md (hunk 9 above)
Comment (highway7) [reply example-workflows/README.md]: Agreed, and removed this step from Setup section.

File: example-workflows/README.md (hunk 10 above)
Comment (highway7) [reply example-workflows/README.md]: Removed the unnecessary IAM roles.

File: example-workflows/README.md (hunk 11 above)
Comment (highway7) [reply example-workflows/README.md:9]: Corrected the steps to align with workflow file.

File: example-workflows/README.md (hunk 9 above)
Comment (averikitsch) [reply example-workflows/README.md]: With this change, for the push to main to kick off the workflow the workflow files needs to be under `.github/workflows/cloud-functions.yml` not `example-workflows/.github/workflows/cloud-functions.yml` and the action needs to be able to find the function in `example-workflows/` using the input `source_dir`.

**Comment by averikitsch** [comment]
One more comment but overall this is looking good! Thanks @highway7

File: example-workflows/README.md (hunk 9 above)
Comment (highway7) [reply example-workflows/README.md]: @averi, if I understand your comments correctly, here is what you are suggesting:  
1. Move example-workflows/.github/workflows/cloud-functions.yml to .github/workflows/deploy-cf-example.yml
2. Change the folder "example-workflows" to "example-app"
3. Use source-dir in deploy-cf-example.yml to find the example app in example-app

Thanks!

File: example-workflows/README.md (hunk 8 above)
Comment (bharathkkb) [reply example-workflows/README.md]: @averikitsch isn't Cloud Build needed?

https://cloud.google.com/functions/docs/building#characteristics_of_the_default_process

File: example-workflows/README.md (hunk 9 above)
Comment (averikitsch) [reply example-workflows/README.md]: Yes, that would be great!

File: example-workflows/README.md (hunk 8 above)
Comment (averikitsch) [reply example-workflows/README.md]: Thanks for catching this. I was not aware of the new update requiring Cloud Build.

File: .gitignore (hunk 3 above)
Comment (averikitsch) [reply .gitignore:1]: Please remove these changes

**Comment by averikitsch** [comment]
@highway7 Please remove the changes to the .gitignore file.

**Comment by highway7** [comment]
> @highway7 Please remove the changes to the .gitignore file.

Sorry about the mess. Reverted .gitignore file back to  9741a48.

**Review by averikitsch — APPROVED** [review]

**Pull request closed without merging**